docs/DeveloperGuide.md

```diff
@@ -1,38 +1,230 @@
+
+**Interaction of components**
+
+<div align="center">
+<img src="UML/Images/overall_sequence-0.png"/>
+<p>
+Figure 1.2
+</p>
+</div>
+
+The sequence diagram as shown in Figure 1.2 shows the interaction of all the main components when a user enters the 
+```start``` command and subsequently enters the ```finish``` command to simulate the process of a workout session 
+experienced by the user.
+
+The five main components in the diagram shows the main classes participating in the process.
+
+These classes will be defined into its own API as an ```interface``` in the upcoming iteration
+
+More details on these components will be described below.
+
+### UI Component
+
+Represented by the Ui class, it handles the interactions between the user and the program. In addition, it also displays
+a basic overview of what the program does.
+
+Types of pre-defined messages:
+
+1. Information messages that describe the functionalities of the program (`Greet` , `Bye`, `PrintExercises` classes)
+2. Help messages that details the usage of the possible commands available in the program (`PrintHelpMessage` class)
+3. Error messages that trigger when a user's input is incorrect and provides an explanation to the
+   user. (`ErrorMessages` class)
+
+### Storage Component
+
+API: [```StorageHandler.java```](https://github.com/AY2223S2-CS2113-W13-2/tp/blob/master/src/main/java/seedu/duke/storage/StorageHandler.java)
+
+The *Storage* component handles the reading and writing of user data to and from the local hard disk in the form of
+a json file.
+
+Key Aspects:
+
+* Handles the serialization and deserialization of user data into a json file using
+  the [Gson Library](https://github.com/google/gson)
+* Handles the creation of user data file when previous one is missing or corrupted
+* Handles the loading of user data and plans upon start of the program
+
+The class diagram as shown in *Figure 1.3* illustrates the structure of the different classes in Storage.
+<div align="center">
+<img src="UML/Images/Storage.png"/>
+<p>
+Figure 1.3
+</p>
+</div>
+
+The StorageHandler interacts with the other classes as shown in the *Sequence Diagram*
+where it shows how the StorageHandler loads the local user data json file as well as the user plans json file upon the 
+resumption of the program.
+<div align="center">
+<img src="UML/Images/LoadingUserCareerData.png"/>
+<p>
+Figure 1.4
+</p>
+</div>
+
+### Command Handler Component
+
+The command handler component consists of multiple states that the program can be at any time. 
+It controls where the user input is being processed.
+
+#### Key Aspects: 
+
+* Handles when there is an exercise ongoing, giving access to finishing and hence saving finished exercises, and keeping track of what exercises there are.
+* Handles when there is no exercise ongoing, granting access to generating new exercises with different confines.
+
+<div align="center">
+<img src="UML/Images/CommandHandler.png"/>
+<p>
+Figure 1.5
+</p>
+</div>

```

Comment:
![image](https://user-images.githubusercontent.com/88542282/228424227-58adfe39-9b87-4d5a-a84d-0677016ea451.png)
Perhaps can consider using reference frame over `ExerciseSessionCommandHandler`/`GeneralCommandHandler` then would also be nice to have separate sequence diagram in the next section(s) to show details.